Add docs for abstract class generators (#39954)
rails/rails: guides/source/active_record_multiple_databases.md

where you will store migrations for that database. We'll look more at `migrations_paths`
later on in this guide.

> Now that we have a new database, let's set up the connection maodel. In order to use the

Reviewer: typo `maodel` -> `model`
 ```suggestion
Now that we have a new database, let's set up the connection model. In order to use the
```

$ bin/rails generate migration CreateDogs name:string --database animals
```

If you are using Rails generators, the scaffold and model generators will create the abstract
class for you. Simply pass the database key to the command line

> ```

Reviewer 2: ```suggestion
```bash
```

$ bin/rails generate scaffold Dog name:title --database animals
```

> And a class with the database name `Animals` and `Record` will be created:

Reviewer 2: I'm not sure if this is actually better, but what do you think about
```suggestion
And a -`Record` class based on the database name `animals` will be created:
```

Author: I incorporated all the other changes except this one because I didn't think it was clearer. Thanks for the suggestion though.

Author: Ended up with a different phrasing on my second pass.

```ruby
class AnimalsRecord < ApplicationRecord
  self.abstract_class = true

  connects_to database: { writing: :animals }
end
```

The generated model will automatically inherit from `AnimalsRecord`.

> ```

Reviewer: ```suggestion
```ruby
```

class Dog < AnimalsRecord
end
```

Note: Since Rails doesn't know which database is the replica for your writer you will need to
add this to the abstract class after you're done.

Rails will only generate the new class once. It will not be overwritten by new scaffolds
or deleted if the scaffold is deleted.

If you already have an abstract class and it's name differs from `AnimalsRecord` you can pass
the `--parent` option to indicate you want a different abstract class:

> ```

Reviewer 2: ```suggestion
```bash
```
